streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -151,7 +156,8 @@ final void transitionTo(final Task.State newState) {
 
     @Override
     public void updateInputPartitions(final Set<TopicPartition> topicPartitions, final Map<String, List<String>> allTopologyNodesToSourceTopics) {
-        this.inputPartitions = topicPartitions;
+        this.inputPartitions.clear();
```

Comment:
Minor cleanup: do deep copy since the passed in `topicPartition` param maybe modified.